providers/common/capabilities.c

```diff
@@ -135,23 +145,28 @@ static const OSSL_PARAM param_group_list[][10] = {
     TLS_GROUP_ENTRY("secp192k1", "secp192k1", "EC", 17),
 #  endif
     TLS_GROUP_ENTRY("secp192r1", "prime192v1", "EC", 18),
+    TLS_GROUP_ENTRY("P-192", "prime192v1", "EC", 18), /* Alias of above */
 #  ifndef FIPS_MODULE
     TLS_GROUP_ENTRY("secp224k1", "secp224k1", "EC", 19),
 #  endif
     TLS_GROUP_ENTRY("secp224r1", "secp224r1", "EC", 20),
+    TLS_GROUP_ENTRY("P-224", "secp224r1", "EC", 20), /* Alias of above */
 #  ifndef FIPS_MODULE
     TLS_GROUP_ENTRY("secp256k1", "secp256k1", "EC", 21),
 #  endif
     TLS_GROUP_ENTRY("secp256r1", "prime256v1", "EC", 22),
+    TLS_GROUP_ENTRY("P-256", "prime256v1", "EC", 22), /* Alias of above */
     TLS_GROUP_ENTRY("secp384r1", "secp384r1", "EC", 23),
+    TLS_GROUP_ENTRY("P-384", "secp384r1", "EC", 23), /* Alias of above */
     TLS_GROUP_ENTRY("secp521r1", "secp521r1", "EC", 24),
+    TLS_GROUP_ENTRY("P-521", "secp521r1", "EC", 24), /* Alias of above */
 #  ifndef FIPS_MODULE
     TLS_GROUP_ENTRY("brainpoolP256r1", "brainpoolP256r1", "EC", 25),
     TLS_GROUP_ENTRY("brainpoolP384r1", "brainpoolP384r1", "EC", 26),
     TLS_GROUP_ENTRY("brainpoolP512r1", "brainpoolP512r1", "EC", 27),
 #  endif
-    TLS_GROUP_ENTRY("x25519", "x25519", "X25519", 28),
-    TLS_GROUP_ENTRY("x448", "x448", "X448", 29),
+    TLS_GROUP_ENTRY("x25519", "X25519", "X25519", 28),
+    TLS_GROUP_ENTRY("x448", "X448", "X448", 29),
```

Comment:
Why this particular change of case for x25519 and x448?

Reply:
This fixes a slight oddity in our x25519/x448 implementation revealed by this PR. In OpenSSL 1.1.1 the group name is "X25519" (i.e. uppercase X). We did not accept the lowercase form. So `openssl s_server -groups X25519` works in 1.1.1, but `openssl s_server -groups x25519` does not. However the IANA registry lists the official group name for TLS as `x25519` (i.e. lowercase x).

The reason we previously had uppercase X is because that's the way the internal OBJ database has it. But when we introduced the "TLS-GROUP" capability the first named parameter is supposed to be the IANA group name, and the 2nd parameter is supposed to be the name that we know it by internally.

Prior to this PR a configuration such as "-groups X25519" would first look up the name in the OBJ database, and if it didn't find it there, then it would check all the groups names from the "TLS-GROUP" capability. Hence both "openssl s_server -groups X25519" and "openssl s_server -groups x25519" both worked in master.

This PR simplifies that code to only run it off the "TLS-GROUP" capability. However it exposed this small discrepancy between the official name and the internal name. This fixes things so that the internal name is the same as how we actually know it internally, and means that "openssl s_server -X25519" continues to work.

Reply:
Thanks for the explanation.